game_patch/multi/server.cpp

```diff
@@ -657,6 +665,26 @@ CodeInjection multi_limbo_init_injection{
     },
 };
 
+CallHook<void(const char* filename)> multi_change_level_call_hook{
+    0x0046E89C,
+    [](const char* filename){
+        if (rf::is_dedicated_server && g_additional_server_config.randomize_rotation) {
+            std::srand(static_cast<unsigned int>(std::time(nullptr)));
+            // select a random level index from the rotation, excluding the currently loaded level
+            int rand_level_index = rf::netgame.current_level_index;
+            int num_levels = rf::netgame.levels.size();
+            do {
+                rand_level_index = std::rand() % num_levels;
+            } while (rand_level_index == rf::netgame.current_level_index && rf::netgame.levels.size() > 1); // prevent infinite loop if rotation has only 1 map
+            rf::netgame.current_level_index = rand_level_index;
+            const char* new_filename = rf::netgame.levels[rf::netgame.current_level_index];
+            multi_change_level_call_hook.call_target(new_filename);
```

Comment:
You repeat `multi_change_level_call_hook.call_target` - this can be easily rewritten to call it once and just change the argument